### PR TITLE
Issue pkg-varnish-cache/issues/139 - Add package for Ubuntu 20.04

### PR DESCRIPTION
**Pull request by martin-g**
https://github.com/varnishcache/pkg-varnish-cache/issues/139

Add packaging jobs for Ubuntu 20.04

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -422,8 +438,10 @@ workflows:
           requires:
             - x64-ubuntu-xenial
             - x64-ubuntu-bionic
+            - x64-ubuntu-focal
             - aarch64-ubuntu-xenial
             - aarch64-ubuntu-bionic
+            - aarch64-ubuntu-focal
             - x64-debian-bullseye
             - aarch64-debian-bullseye
             - x64-debian-buster
```
Comment (Dridi) [inline .circleci/config.yml:447]: It's curious that `debian` jobs are ordered by version first and architecture second whereas everything else does the opposite.

Can you please also align `debian` entries with the rest?

Comment (martin-g) [reply .circleci/config.yml]: I've re-ordered them by oldest to newest version and grouped them first `x64` followed by its `aarch64` twin.

**Comment by gquintard** [comment]
merged, thanks @martin-g

**Comment by 7php** [comment]
Awesome!
When can we expect a package for `focal` on https://packagecloud.io/?

**Comment by gquintard** [comment]
weeklies are built on mondays and if the team ok's them, they are pushed to packagecloud on the same day

**Comment by martin-g** [comment]
I was going to suggest to download the nightly built packages.tar.gz from CircleCI but there is no execution of the `nightly` workflow for some reason.
There are several failed builds for the PRs of @andrewwiik because his branches are "This branch is 753 commits ahead, 2312 commits behind varnishcache:master." but I don't think this is related to the missing `nightly` build. I'll try to investigate!

**Comment by martin-g** [comment]
The job just kicked!
It just became 4 AM UTC! :-)
I'm going for a coffee!

@7php once https://app.circleci.com/pipelines/github/varnishcache/varnish-cache/527/workflows/86759a67-825d-448e-9aae-2f2a0e4ba014 finishes you will be able to download `packages.tar.gz` from the `collect_packages` job's artifacts. Inside you will find a folder for Ubuntu `focal`

**Comment by martin-g** [comment]
Unfortunately the build of `focal` failed

```
mk-build-deps --install debian/control
 
semop(1): encountered an error: Function not implemented
 
Error in the build process: exit status 1
 
dpkg: error: cannot access archive 'varnish-build-deps_20200516-weekly~focal-1_all.deb': No such file or directory
 
mk-build-deps: dpkg --unpack failed
 
+ true
 
+ echo 'Build the packages...'
 
Build the packages...
 
+ dpkg-buildpackage -us -uc -j16
 
dpkg-buildpackage: info: source package varnish
 
dpkg-buildpackage: info: source version 20200516-weekly~focal-1
 
dpkg-buildpackage: info: source distribution UNRELEASED
 
dpkg-buildpackage: info: source changed by Denis Braekhus <denis@varnish-software.com>
 
dpkg-buildpackage: info: host architecture arm64
 
 dpkg-source --before-build .
 
dpkg-checkbuilddeps: error: Unmet build dependencies: dh-systemd libedit-dev libjemalloc-dev libncurses-dev libpcre3-dev python3-docutils python3-sphinx
 
dpkg-buildpackage: warning: build dependencies/conflicts unsatisfied; aborting
 
dpkg-buildpackage: warning: (Use -d flag to override.)
 ```

https://app.circleci.com/pipelines/github/varnishcache/varnish-cache/527/workflows/86759a67-825d-448e-9aae-2f2a0e4ba014/jobs/6009

**Comment by 7php** [comment]
oh noooooo!!! 
But I really appreciate the effort @martin-g !!!
I nearly jumped out of my chair screaming 'hurayy' until I see this next notification.. heheh!

**Comment by martin-g** [comment]
It failed only for `aarch64`. The build for `x86_64` passed!
But `collect_packages` won't execute until we either fix the issue for `aarch64` or exclude it as dependency.

@gquintard @Dridi I am not sure whether I'll have time to investigate before Monday! Feel free to disable `aarch64-ubuntu-focal` job if you need packages.tar.gz at Monday!

**Comment by 7php** [comment]
having an `x86_64` for monday would be cool.
Thank you for all the hard effort @martin-g , you are awesome!
Have a good weekend!

**Comment by martin-g** [comment]
https://github.com/varnishcache/varnish-cache/pull/3328

**Pull request merged**